packages/common/src/db/queries/Candidate.ts

```diff
@@ -1605,3 +1605,45 @@ export const setValid = async (
     },
   ).exec();
 };
+
+export const getUniqueNameSet = async (): Promise<any> => {
+  const nameSet = new Set();
+  const allNodes = await allCandidates();
+  for (const node of allNodes) {
+    nameSet.add(node.name);
+  }
+  return Array.from(nameSet);
+};
+
+export const getDuplicatesByName = async (): Promise<any> => {
+  const duplicates = [];
+  const names = await getUniqueNameSet();
+  for (const name of names) {
+    const candidates = await CandidateModel.find({ name: name }).exec();
```

Comment:
can you make all these typed instead of returning any ? 

i.e.
await CandidateModel.find<CandidateData>({ name: name }).exec();